core/src/apps/bitcoin/scripts.py

```diff
@@ -238,6 +239,10 @@ def output_script_native_segwit(witver: int, witprog: bytes) -> bytearray:
     length = len(witprog)
     utils.ensure((length == 20 and witver == 0) or length == 32)
 
+    # check that P2TR address encodes a valid BIP340 public key
+    if witver == 1:
+        utils.ensure(bip340.verify_publickey(witprog))
```

Comment:
but maybe not here? istm this is not the right place for the check anyway, it should come at the address decoding step imo

Reply:
I agree about raising `wire.DataError`. Technically is seems to make more sense to catch this in `common.decode_bech32_address()` rather than here, but doesn't make any difference at the moment, so I have no strong preference either way.